wear/src/main/java/com/eveningoutpost/dexdrip/Services/CustomComplicationProviderService.java

```diff
@@ -129,34 +130,56 @@ public void onComplicationUpdate(
                 is_stale = true;
             }
         }
+
         Log.d(TAG, "Returning complication text: " + numberText);
 
         COMPLICATION_STATE state = COMPLICATION_STATE.get_enum((int) PersistentStore.getLong(ComplicationTapBroadcastReceiver.COMPLICATION_STORE));
         if (state == null) state = COMPLICATION_STATE.DELTA;
 
         ComplicationData complicationData = null;
 
-        final boolean doMgdl = Home.getPreferencesStringWithDefault("units", "mgdl").equals("mgdl");
         switch (dataType) {
             case ComplicationData.TYPE_SHORT_TEXT:
 
                 final ComplicationData.Builder builder = new ComplicationData.Builder(ComplicationData.TYPE_SHORT_TEXT)
                         .setShortText(ComplicationText.plainText(numberText))
                         .setTapAction(complicationPendingIntent);
-                UserError.Log.d(TAG, "Current complication state:" + state);
+
+                UserError.Log.d(TAG, "TYPE_SHORT_TEXT Current complication state:" + state);
                 switch (state) {
                     case DELTA:
-                        builder.setShortTitle(!is_stale ? (ComplicationText.plainText(bgReading != null ? unitizedDeltaString(false, false, Home.get_follower(), doMgdl) : "null")) : ComplicationText.plainText(""));
+                        builder.setShortTitle(ComplicationText.plainText(getDeltaText(bgReading, is_stale)));
                         break;
                     case AGO:
                         builder.setShortTitle(ComplicationText.plainText(niceTimeSinceBgReading(bgReading)));
+                        //Loop status by @gregorybel
+                        String externalStatusString = PersistentStore.getString("remote-status-string");
+                        Log.d(TAG, "Returning complication status: " + externalStatusString);
```

Comment:
This is going to negatively impact a large portion of users because it overwrites the `.setShortText` with loop status even if none is present. It should at least be conditional on there being any persistently stored status message and revert to the old behavior if none is set.